tests/Goteo/Repository/DataSetRepositoryTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Goteo\Repository\Tests;
+
+use Goteo\Application\Config;
+use Goteo\Application\Exception\ModelNotFoundException;
+use Goteo\Core\Model;
+use Goteo\TestCase;
+use Goteo\Model\DataSet;
+use Goteo\Repository\DataSetRepository;
+
+class DataSetRepositoryTest extends TestCase
+{
+
+    private DataSetRepository $repository;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->repository = new DataSetRepository();
+    }
+
+    public function testExceptionOnGetById(): void {
+
+        $id = 1;
+        $this->expectException(ModelNotFoundException::class);
+        $this->expectExceptionMessage("DataSet with id {$id} not found");
+
+        $this->repository->getById($id);
+
+    }
+
+    public function testGetList(): void {
+        $dataSets = $this->repository->getList([]);
+
+        $this->assertIsArray($dataSets);
+        $this->assertCount(0, $dataSets);
+    }
+
+    /**
+     * @depends testGetList
+     */
+    public function testSaveDataSet(): DataSet {
+        $dataSet = new DataSet();
+        $dataSet->setId(1)
+            ->setTitle('Data Set 1.')
+            ->setDescription(' Description of DataSet 1')
+            ->setUrl(Config::get('url.main'));
```

Comment:
Let's avoid calling static methods as much as possible. We can simply write "http://www.example.com" as an URL.